service/s3/s3crypto/hash_io.go

```diff
@@ -0,0 +1,61 @@
+package s3crypto
+
+import (
+	"crypto/md5"
+	"crypto/sha256"
+	"hash"
+	"io"
+)
+
+// HashReader is used for calculating SHA256 when following the sigv4 specification
+// and the unencrypted MD5.
+type HashReader interface {
+	GetValue() []byte
+	GetContentLength() int
```

Comment:
Should `GetContentLength` be a `int64`? `int` on 32 bit systems will limit the content to 2GB.